compiler/rustc_borrowck/src/lib.rs

```diff
@@ -193,9 +193,7 @@ fn do_mir_borrowck<'tcx>(
         .map(|(idx, body)| (idx, MoveData::gather_moves(body, tcx, |_| true)));
 
     let mut flow_inits = MaybeInitializedPlaces::new(tcx, body, &move_data)
-        .into_engine(tcx, body)
-        .pass_name("borrowck")
-        .iterate_to_fixpoint()
+        .iterate_to_fixpoint(tcx, body, Some("borrowck"))
```

Comment:
Further simplification: should we always give a name and get rid of the option?

Reply:
I just took a look. The optional pass name is unwrapped as `"-----"` here:

https://github.com/rust-lang/rust/blob/1e4f10ba6476e48a42a79b9f846a2d9366525b9e/compiler/rustc_mir_dataflow/src/framework/engine.rs#L215

So everywhere I've used `None` would be changed to `"-----"` which would be weird. More generally, the file naming scheme for IR dumping is complicated. I don't want to delve into that for this PR, which is a straightforward refactoring. Maybe in a follow-up.